src/main/java/net/coderbot/iris/pipeline/newshader/fallback/ShaderSynthesizer.java

```diff
@@ -166,9 +168,13 @@ public static String fsh(ShaderAttributeInputs inputs, FogMode fogMode, AlphaTes
 				main.append(".rrrr");
 			}
 
-			main.append(" * vertexColor;\n");
+			if (alphaTest == AlphaTests.VERTEX_ALPHA) {
+				main.append(" * vec4(iris_vertexColor.rgb, 1);\n");
+			} else {
+				main.append(" * iris_vertexColor;\n");
+			}
 		} else {
-			main.append("    vec4 color = vertexColor;\n");
+			main.append("    vec4 color = iris_vertexColor;\n");
```

Comment:
Though this case doesn't yet happen in vanilla, strictly speaking there's nothing preventing per-vertex alpha without a texture. It would just be really silly. Might as well put the logic here too.